### PR TITLE
RN generators improvements

### PR DESCRIPTION
**Pull request by adi88d**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25940
https://github.com/demisto/etc/issues/25752

## Description
- Use ValidateManager instead of FilesValidator to gets the modified files.
- Write the file description in to the RN if the file is new.
- Update RN format for file types without description: Connections, Incident Types, Indicator Types, Layouts, Incident Fields.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/validate/validate_manager.py [hunk 1]
```diff
@@ -1023,3 +1025,15 @@ def get_packs_that_should_have_version_raised(modified_files, changed_meta_packs
...
         return changed_meta_packs.union(modified_packs_that_should_have_version_raised)
+
+    @staticmethod
+    def get_packs(changed_files):
+        packs = set()
+        for changed_file in changed_files:
+            if isinstance(changed_file, tuple):
+                changed_file = changed_file[1]
+            pack = get_pack_name(changed_file)
+            if pack and is_file_path_in_pack(changed_file):
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/validate_manager.py:1036]: why do you need to call `is_file_path_in_pack`?
if the file is not in a pack, `pack` will be None.

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 2]
```diff
@@ -25,11 +25,11 @@
 AZURE_FEED_PACK_PATH = join(TEST_FILES_PATH, "content_repo_example/Packs/FeedAzure")
 AZURE_FEED_INVALID_PACK_PATH = join(TEST_FILES_PATH, "content_repo_example/Packs/FeedAzureab")
 VALID_PACK_PATH = join(TEST_FILES_PATH, "content_repo_example/Packs/FeedAzureValid")
-VALID_PLAYBOOK_FILE_PATH = 'demisto_sdk/tests/test_files/Packs/CortexXDR/Playbooks/Cortex_XDR_Incident_Handling.yml'
-INVALID_PLAYBOOK_FILE_PATH = 'demisto_sdk/tests/test_files/Packs/CortexXDR/Playbooks/' \
-                             'Cortex_XDR_Incident_Handling_invalid.yml'
-VALID_SCRIPT_PATH = 'demisto_sdk/tests/test_files/Packs/' \
-                    'CortexXDR/Scripts/EntryWidgetNumberHostsXDR/EntryWidgetNumberHostsXDR.yml'
+VALID_PLAYBOOK_FILE_PATH = join(TEST_FILES_PATH, "Packs/CortexXDR/Playbooks/Cortex_XDR_Incident_Handling.yml")
+INVALID_PLAYBOOK_FILE_PATH = join(TEST_FILES_PATH, "Packs/CortexXDR/Playbooks/Cortex_XDR_Incident_Handling_invalid.yml")
```
Comment (yaakovi) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:29]: as long as you are updating it to use join, please fix all of them...
```suggestion
INVALID_PLAYBOOK_FILE_PATH = join(TEST_FILES_PATH, 'Packs', 'CortexXDR', 'Playbooks', 'Cortex_XDR_Incident_Handling_invalid.yml')
```

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 3]
```diff
@@ -363,3 +386,22 @@ def create_markdown(self, release_notes_path: str, rn_string: str, changed_files
             self.existing_rn_changed = True
             with open(release_notes_path, 'w') as fp:
                 fp.write(rn_string)
+
+
+def get_file_description(path, file_type):
+    if not os.path.isfile(path):
+        print_error(f'Gets file description failed: "{path}" file does not exist')
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:393]:
```suggestion
        print_warning(f'Cannot get file description: "{path}" file does not exist')
```

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 4]
```diff
@@ -363,3 +386,22 @@ def create_markdown(self, release_notes_path: str, rn_string: str, changed_files
...
+        return yml_file.get('description', '')
+
+    elif file_type == 'Script':
+        yml_file = get_yaml(path)
+        return yml_file.get('comment', '')
+
+    elif file_type in ('Classifiers', 'Reports', 'Widgets', 'Dashboards'):
+        json_file = get_json(path)
+        return json_file.get('description', '')
+    return '%%UPDATE_RN%%'
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:407]: add an empty line before last `return`

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 5]
```diff
@@ -255,84 +257,105 @@ def build_rn_template(self, changed_items: dict):
...
                 if not connections_header:
                     rn_string += '\n#### Connections\n'
                     connections_header = True
-                rn_string += f'##### {k}\n- %%UPDATE_RN%%\n'
+                rn_string += rn_desc
         return rn_string
 
     def update_existing_rn(self, current_rn, changed_files):
         new_rn = current_rn
         for k, v in sorted(changed_files.items(), reverse=True):
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:338]: please switch `k, v` to meaning full names (at least `k`)

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 6]
```diff
@@ -255,84 +257,105 @@ def build_rn_template(self, changed_items: dict):
...
+            desc = v.get('description', '')
+            v = v.get('type', '')
+
             if v is None:
                 continue
+
+            if v in ('Connections', 'Incident Types', 'Indicator Types', 'Layouts', 'Incident Fields'):
+                rn_desc = f'\n- **{k}**'
+            else:
+                rn_desc = f'\n##### {k}\n- {desc}\n' if is_new_file else f'\n##### {k}\n- %%UPDATE_RN%%\n'
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:349]:
```suggestion
                rn_desc = f'\n##### New: {k}\n- {desc}\n' if is_new_file else f'\n##### {k}\n- %%UPDATE_RN%%\n'
```

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 7]
```diff
@@ -58,7 +58,9 @@ def execute_update(self):
             self.find_added_pack_files()
             for packfile in self.pack_files:
                 file_name, file_type = self.identify_changed_file_type(packfile)
-                changed_files[file_name] = file_type
+                changed_files[file_name] = {'type': file_type, 'description': get_file_description(packfile, file_type)}
+                changed_files[file_name]['is_new_file'] = True if packfile in self.added_files else False
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:62]:
```suggestion
                changed_files[file_name] = {
                    'type': file_type,
                    'description': get_file_description(packfile, file_type),
                    'is_new_file': packfile in self.added_files,
                }
```

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 8]
```diff
@@ -257,76 +261,77 @@ def build_rn_template(self, changed_items: dict):
...
+            if not _type:
                 continue
 
-            if v in ('Connections', 'Incident Types', 'Indicator Types', 'Layouts', 'Incident Fields'):
-                rn_desc = f'- **{k}**\n'
+            if _type in ('Connections', 'Incident Types', 'Indicator Types', 'Layouts', 'Incident Fields'):
+                rn_desc = f'- **{content_name}**\n'
             else:
-                rn_desc = f'##### {k}\n- {desc}\n' if is_new_file else f'##### {k}\n- %%UPDATE_RN%%\n'
+                rn_desc = f'##### {content_name}\n- {desc}\n' if is_new_file \
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:274]: missed that one the first time...
```suggestion
                rn_desc = f'##### New: {content_name}\n- {desc}\n' if is_new_file \
```

**Comment by bakatzir** [comment]
probably fixes: https://github.com/demisto/etc/issues/26858

**Pull request merged**